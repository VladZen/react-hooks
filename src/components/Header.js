import PropTypes from 'prop-types'

const Header = ({ heading, children }) => {
  const buttons = children
    ? <div className='layout__header__buttons'>{ children }</div>
    : ''

  return (
    <header className='layout__header'>
      { buttons }
      <h1
        className='text-ellipsis'
        title={ heading }
      >
        { heading }
      </h1>
    </header>
  )
}

Header.defaultProps = {
  heading: ''
}

Header.propTypes = {
  heading: PropTypes.string
}

export default Header
