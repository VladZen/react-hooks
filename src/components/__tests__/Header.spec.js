import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import Header from '@/components/Header'

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Header component', () => {
  describe('rendering', () => {
    it('without children', () => {
      act(() => {
        render(<Header heading='Some title' />, container)
      })

      expect(container).toMatchSnapshot()
    })

    it('with children', () => {
      act(() => {
        render((<Header heading='Some title'>
          <a href='#'>
            Some link
          </a>
        </Header>), container)
      })

      expect(container).toMatchSnapshot()
    })
  })
})

