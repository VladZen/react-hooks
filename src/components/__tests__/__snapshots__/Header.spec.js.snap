// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component rendering with children 1`] = `
<div>
  <header
    class="layout__header"
  >
    <div
      class="layout__header__buttons"
    >
      <a
        href="#"
      >
        Some link
      </a>
    </div>
    <h1
      class="text-ellipsis"
      title="Some title"
    >
      Some title
    </h1>
  </header>
</div>
`;

exports[`Header component rendering without children 1`] = `
<div>
  <header
    class="layout__header"
  >
    
    <h1
      class="text-ellipsis"
      title="Some title"
    >
      Some title
    </h1>
  </header>
</div>
`;
